File: Ridy-taxi-app/android/common/src/main/java/com/innomalist/taxi/common/models/Transaction.kt
package com.innomalist.taxi.common.models

import com.squareup.moshi.JsonClass
import java.sql.Timestamp
import java.text.SimpleDateFormat
import java.util.*

@JsonClass(generateAdapter = true)
data class Transaction(
        var amount: Double,
        val currency: String,
        var documentNumber: String?,
        var transactionTime: Long,
        var details: String?,
        var id: Long,
        var transactionType: String
) {
    val day: String
        get() {
            val date = Date(transactionTime)
            return SimpleDateFormat("dd", Locale.getDefault()).format(date)
        }
    val month: String
        get() {
            val date = Date(transactionTime)
            return SimpleDateFormat("MMM", Locale.getDefault()).format(date)
        }
}

File: Ridy-taxi-app/ios/common/Models/Objects/Transaction.swift
//
//  Transaction.swift
//
//  Created by Minimalistic Apps on 11/15/18
//  Copyright (c) . All rights reserved.
//

import Foundation

public final class Transaction: Codable {

  public var documentNumber: String?
  public var id: Int?
  public var transactionTime: Double?
  public var amount: Double?
  public var currency: String?
  public var transactionType: String?
}
